x/auth/types/expected_keepers.go

```diff
@@ -19,4 +19,6 @@ type BankKeeper interface {
 type AccountsModKeeper interface {
 	SendModuleMessageUntyped(ctx context.Context, sender []byte, msg protoiface.MessageV1) (protoiface.MessageV1, error)
 	IsAccountsModuleAccount(ctx context.Context, accountAddr []byte) bool
+	NextAccountNumber(ctx context.Context) (accNum uint64, err error)
+	SetAccountNumber(ctx context.Context, currentAccNum uint64) error
```

Comment:
personal preference: the SetAccountNumber is used for migration only. Maybe this can go into a new interface that is imported to make it more verbose or some godoc can be hepful.